meteor-app/imports/ui/pages/company-search-trial.jsx

```diff
@@ -50,23 +50,25 @@ const SearchResults = ({ searchText }) => (
 				return <h2>{`Error! ${error.message}`}</h2>;
 			}
 
-			const resultList = data.searchCompanies.map(function(company) {
-				// if (Meteor.isDevelopment) {
-				// 	console.log("BEFORE PROCESS");
-				// 	console.log(company.locations);
-				// }
-				// const newLocations = company.locations.map(location =>
-				// 	processLocation(JSON.stringify(location))
-				// );
-				// if (Meteor.isDevelopment) {
-				// 	console.log("newLocations");
-				// 	console.log(newLocations);
-				// }
-				// company.locations = newLocations;
-				// if (Meteor.isDevelopment) {
-				// 	console.log("AFTER PROCESS");
-				// 	console.log(company.locations);
-				// }
+			// searchCompanies is read-only, we do a
+			// deep copy before we mutate it with sort:
+			// https://stackoverflow.com/questions/597588/how-do-you-clone-an-array-of-objects-in-javascript
+			const resultList = data.searchCompanies
+			.map(c => Object.assign({}, c))
+			// Array.prototype.sort is in-place and returns the new array
+			.sort(function(a, b) {
+				// This scoring method was given to me by Krit,
+				// who told me that Julian wanted it this way.
+				const score = (company) => (company.numJobAds * 2) + (company.numReviews * 1.5) + (company.numSalaries * 1);
+				const aScore = score(a);
```

Comment:
Only thing I think I'm not understanding is why there is an aScore and a bScore. What is the difference? I imagine that a and b are both two companies?

Reply:
Okay...
Notice that they occur as part of the argument to Array.prototype.sort. sort takes a sorting function as an argument. The docs explain what arguments the sorting function can take and what values it should return in order for the array to be sorted "properly". This allows you to sort arrays however you want.
So sort takes an argument (optionally) which is a function. That function which it takes as an argument, the "sorting function", is supposed to take two arguments: the items being compared. Now at any given point when the sorting function is called, its arguments could theoretically be any pair of items taken from the array being sorted. Lucky for you, in this case all the items in the array are companies.
So how are we specifically here at Vize wanting to determine which company comes first in the results list? Whichever one has the higher "score". The "scoring algorithm" was given to me by @krit-gpt , who says he got it from you.
Does that all make sense?

Reply:
There are actually links to some relevant documentation in the pull request description...

Reply:
Okay awesome, thanks for taking the time to explain that makes sense, I had a feeling the two scores would be used to be compared against each other. I'll take a look at some of the documentation as well to understand how the sort function works more in depth.